dao/src/test/java/org/thingsboard/server/dao/sql/user/JpaUserCredentialsDaoTest.java

```diff
@@ -34,41 +38,73 @@
  */
 public class JpaUserCredentialsDaoTest extends AbstractJpaDaoTest {
 
+    public static final String ACTIVATE_TOKEN = "ACTIVATE_TOKEN_0";
+    public static final String RESET_TOKEN = "RESET_TOKEN_0";
+    final int COUNT_USER_CREDENTIALS = 2;
+    List<UserCredentials> userCredentialsList;
+
     @Autowired
     private UserCredentialsDao userCredentialsDao;
 
+    @Before
+    public void setUp() {
+        userCredentialsList = new ArrayList<>();
+        for (int i=0; i<COUNT_USER_CREDENTIALS; i++) {
+            userCredentialsList.add(createUserCredentials(i));
+        }
+    }
+
+    UserCredentials createUserCredentials(int number) {
+        UserCredentials userCredentials = new UserCredentials();
+        userCredentials.setEnabled(true);
+        userCredentials.setUserId(new UserId(UUID.randomUUID()));
+        userCredentials.setPassword("password");
+        userCredentials.setActivateToken("ACTIVATE_TOKEN_" + number);
+        userCredentials.setResetToken("RESET_TOKEN_" + number);
+        return userCredentialsDao.save(SYSTEM_TENANT_ID, userCredentials);
+    }
+
+    @After
+    public void after() {
+        for (UserCredentials userCredentials : userCredentialsList) {
+            userCredentialsDao.removeById(TenantId.SYS_TENANT_ID, userCredentials.getUuidId());
+        }
+    }
+
     @Test
-    @DatabaseSetup("classpath:dbunit/user_credentials.xml")
     public void testFindAll() {
         List<UserCredentials> userCredentials = userCredentialsDao.find(SYSTEM_TENANT_ID);
-        assertEquals(2, userCredentials.size());
+        assertEquals(COUNT_USER_CREDENTIALS + 1, userCredentials.size());
     }
 
     @Test
-    @DatabaseSetup("classpath:dbunit/user_credentials.xml")
     public void testFindByUserId() {
-        UserCredentials userCredentials = userCredentialsDao.findByUserId(SYSTEM_TENANT_ID, UUID.fromString("787827e6-27d7-11e7-93ae-92361f002671"));
+        UserCredentials result = userCredentialsList.get(0);
+        assertNotNull(result);
+        UserCredentials userCredentials = userCredentialsDao.findByUserId(SYSTEM_TENANT_ID, result.getUserId().getId());
         assertNotNull(userCredentials);
-        assertEquals("4b9e010c-27d5-11e7-93ae-92361f002671", userCredentials.getId().toString());
-        assertEquals(true, userCredentials.isEnabled());
-        assertEquals("password", userCredentials.getPassword());
-        assertEquals("ACTIVATE_TOKEN_2", userCredentials.getActivateToken());
-        assertEquals("RESET_TOKEN_2", userCredentials.getResetToken());
+        assertEquals(result.getId(), userCredentials.getId());
```

Comment:
that is a wrong approach to compare field by field. you should compare objects with right equals and hashcode functions